[Bug] v1.5.10 & v1.6.10 versions are using protobuf > 5 in prod

### Is this a new bug in dbt-core?

- [X] I believe this is a new bug in dbt-core
- [X] I have searched the existing issues, and I could not find an existing issue for this bug

### Current Behavior

[community Slack #adapter-ecosystem thread](https://getdbt.slack.com/archives/C030A0UF5LM/p1711466741203999)

As a part of #9630 PR, we restricted the protobuf to 4.x versions but the manual backport on v1.5 (#9676) & v1.6 (#9675) branches were incorrect because the restriction is only applied to dev-requirements rather than at the setup.py. As we can see in [v1.5.10](https://github.com/dbt-labs/dbt-core/blob/v1.5.10/core/setup.py#L67) and [v1.6.10](https://github.com/dbt-labs/dbt-core/blob/v1.6.10/core/setup.py#L82) This resulted in breaking the downstream adapters compatibilities as they started including protobuf > 5.x which is not compatible with dbt-core. It doesn't surface in dbt-core tests because dev-requirements always ensure that protobuf 4.x is installed. 
Eg for errors happen in adapters 
```
 def msg_to_dict(msg: EventMsg) -> dict:
        msg_dict = dict()
        try:
>           msg_dict = MessageToDict(
                msg, preserving_proto_field_name=True, including_default_value_fields=True  # type: ignore
            )
E           TypeError: MessageToDict() got an unexpected keyword argument 'including_default_value_fields'

/dbt/events/functions.py:227: TypeError
```

Another concern, we should not have duplicate dependencies listed in setup.py and dev-requirements.txt because these manual mistakes can happen again in the future.

### Expected Behavior

All the downstream adapters in v1.5.10 & v1.6.10 should be using the protobuf 4.x versions to avoid the above errors. 

### Steps To Reproduce

1. Use any adapter with latest v1.5 or v1.6 branches eg dbt-core v1.5.10
2. Install the dependencies
3. Check in the pip list if protobuf > 5. If it is then it is not compatible with dbt-core. 

### Relevant log output

_No response_

### Environment

```markdown
- OS: Ubuntu 20.04
- Python:3.8.16
- dbt:1.5
```


### Which database adapter are you using with dbt?

other (mention it in "Additional Context")

### Additional Context

_No response_

--- a/.changes/unreleased/Dependencies-20240331-103917.yaml
+++ b/.changes/unreleased/Dependencies-20240331-103917.yaml
@@ -1,6 +1,6 @@
 kind: Dependencies
-body: Remove unwanted dependency of protobuf in dev-requirements
+body: Remove duplicate dependency of protobuf in dev-requirements
 time: 2024-03-31T10:39:17.432017-07:00
 custom:
   Author: niteshy
-  PR: "9838"
+  Issue: "9830"

--- a/.changes/unreleased/Dependencies-20240331-103917.yaml
+++ b/.changes/unreleased/Dependencies-20240331-103917.yaml
@@ -1,0 +1,6 @@
+kind: Dependencies
+body: Remove unwanted dependency of protobuf in dev-requirements
+time: 2024-03-31T10:39:17.432017-07:00
+custom:
+  Author: niteshy
+  PR: "9838"

--- a/dev-requirements.txt
+++ b/dev-requirements.txt
@@ -14,7 +14,6 @@ ipdb
 mypy==1.4.1
 pip-tools
 pre-commit
-protobuf>=4.0.0,<5.0.0
 pytest>=7.4,<8.0
 pytest-cov
 pytest-csv>=3.0,<4.0

